python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -23,20 +23,14 @@ class ExposureTrackedBuffer(Buffer):
         The size of the slice (in bytes)
     """
 
-    _owner: BufferOwner
-
     def __init__(
         self,
         owner: BufferOwner,
         offset: int = 0,
         size: Optional[int] = None,
     ) -> None:
         super().__init__(owner=owner, offset=offset, size=size)
-        self._owner._slices.add(self)
-
-    @property
-    def exposed(self) -> bool:
-        return self._owner.exposed
+        self.owner._slices.add(self)
```

Comment:
I guess, we could add `assert isinstance(owner, BufferOwner)` in `Buffer.__init__` but I am not sure that is is that pythonic?

Reply:
An assertion would basically be there to protect against developer error. There's nothing unpythonic about it, that's definitely what the functionality is there for. The main question I would have is, how would we end up with something other than a BufferOwner here, and how bad would the failure mode be? Unless the assertion makes it markedly easier to debug the code it doesn't add a whole lot of value IMHO.

Reply:
I think it is arguable a little unpythonic to prevent duck-typing by using `isinstance` to force a specify type (as opposed to a general ABC/protocol type). 
Anyways, in this case `as_buffer()` is the only one creating `ExposureTrackedBuffer` and it explicitly sets the `owner` to `SpillableBufferOwner | SpillableBuffer | BufferOwner` so I don't think a type error is likely here.